### DISCUSSION TITLE
Error: error: 'Panel_GC9A01A' in namespace 'lgfx' does not name a type    lgfx::Panel_GC9A01A _panel_instance;

### DISCUSSION THREAD

**Discussion by oyasai-edu** [lovyan03/LovyanGFX#479] [Q&A]
I have Error that is "error: 'Panel_GC9A01A' in namespace 'lgfx' does not name a type
   lgfx::Panel_GC9A01A _panel_instance;".

Could you give me some hints on what I should check and what I should do?

The source code is as follows.

--------------

```cpp 

#include <LovyanGFX.hpp>

// Seeed Studio XIAO用の設定
class LGFX : public lgfx::LGFX_Device {
  lgfx::Panel_GC9A01A _panel_instance;
  lgfx::Bus_SPI _bus_instance;
public:
  LGFX(void) {
    {
      auto cfg = _bus_instance.config();

      cfg.spi_host = SPI2_HOST; // XIAOではSPI2_HOSTを使用
      cfg.spi_mode = 0;
      cfg.freq_write = 40000000;
      cfg.freq_read = 16000000;
      cfg.spi_3wire = false;
      cfg.use_lock = true;
      cfg.dma_channel = 1;
      cfg.pin_sclk = 13;  // SCLKピン
      cfg.pin_mosi = 11;  // MOSIピン
      cfg.pin_miso = -1;  // MISOは使用しない場合は-1
      cfg.pin_dc = 10;    // DCピン
      _bus_instance.config(cfg);
      _panel_instance.setBus(&_bus_instance);
    }
    {
      auto cfg = _panel_instance.config();
      cfg.pin_cs = 9;     // CSピン
      cfg.pin_rst = 8;    // RSTピン
      cfg.pin_busy = -1;  // BUSYは使用しない場合は-1
      _panel_instance.config(cfg);
    }
    setPanel(&_panel_instance);
  }
};

LGFX lcd;

void setup() {
  lcd.init();
  lcd.setTextColor(LCD_WHITE);
  lcd.setFontSize(2);
  lcd.clear();
  lcd.drawString("こんにちは、世界！", 10, 10);
}

void loop() {
  // ループ内処理
}


```

**Comment by tobozo** [comment]
hi,

the error message is because `Panel_GC9A01A` is not in the list of [supported panels](https://github.com/lovyan03/LovyanGFX/tree/master/src/lgfx/v1/panel). 

if none of the supported panels drivers work, you will probably have to write your own panel driver, [this pull request](https://github.com/lovyan03/LovyanGFX/pull/478/files) is a good contribution example for a new panel.

**Reply by oyasai-edu** [reply]
Thank you for the information. but. The file exists. Even if I import this directly, the same error occurs. Why...

**Reply by tobozo** [reply]
> The file exists.

where does it exist? I don't see it in LovyanGFX source tree, and I cannot guess what is in that file.

if you have a file where `class Panel_GC9A01A` is declared, please share it

**Reply by oyasai-edu** [reply]
Thank you for your reply. The file can be found here.
https://github.com/lovyan03/LovyanGFX/blob/master/src/lgfx/v1/panel/Panel_GC9A01.hpp

**Reply by tobozo** [reply]
thanks,

the name of the class is `Panel_GC9A01`, not `Panel_GC9A01A`

if you edit your configuration code the error should go away:

```cpp
  lgfx::Panel_GC9A01 _panel_instance;
```

**Reply by oyasai-edu** [reply]
Wow, did I make that mistake? Thank you for noticing.
